logic/atrium-logic-common/src/main/kotlin/ch/tutteli/atrium/logic/creating/basic/contains/creators/impl/ContainsAssertionCreator.kt

```diff
@@ -83,9 +89,23 @@ abstract class ContainsAssertionCreator<T : Any, TT : Any, in SC, C : Contains.C
 
     private fun featureFactory(count: Int, numberOfOccurrences: Translatable): AssertionGroup {
         val assertions = checkers.map { it.createAssertion(count) }
-        return assertionBuilder.feature
-            .withDescriptionAndRepresentation(numberOfOccurrences, Text(count.toString()))
-            .withAssertions(assertions)
-            .build()
+        val checker = checkers.firstOrNull()
+        return if (checkers.size == 1 && checker is AtLeastChecker && checker.times == 1) {
+            assertionBuilder.explanatoryGroup
+                .withDefaultType
+                .withAssertion(
+                    assertionBuilder.explanatory
+                        .withExplanation(descriptionNotFound)
+                        .build()
+                ).let {
+                    if (checker.createAssertion(count).holds()) it
+                    else it.failing
+                }.build()
```

Comment:
I'm hesitant to move the `if` out of the featureFactory and into the `searchAndCreateAssertion` methods since it seems like the featureFactory bears the responsibility for assertions about the number of occurrences.

Would it be too much of a hack to return an empty `AssertionGroup` here in the event that `checker.createAssertion(count).holds()` is true? Or alternatively return a passing descriptive assertion `number of occurrences found: N`.

Reply:
Good thoughts, I would change feature Factory to either  `AssertionGroup?` or `Option<AssertionGroup>` (I numbered your options, so either 2 or 3, up to you. The option 1 is a no go IMO). 
And I would realise option 5 in addition, but it's a nice bonus IMO, could be in a subsequent PR.